### PR TITLE
MacOS 13

### PR DESCRIPTION
**Pull request by hannesa2**
It's also somehow a preparation for an arm64 runner

### CODE REVIEW

File: script/update_libgit2 [hunk 1]
```diff
@@ -5,18 +5,19 @@ set -e
 # augment path to help it find cmake installed in /usr/local/bin,
 # e.g. via brew. Xcode's Run Script phase doesn't seem to honor
 # ~/.MacOSX/environment.plist
-PATH="/usr/local/bin:$PATH"
+if [[ `uname -m` == 'arm64' ]]; then
+  echo "** Running on a Apple Silicon M1"
+  PATH="/usr/homebrew/bin:$PATH"
```
Comment (DivineDominion) [inline script/update_libgit2:10]: IIRC it's `/opt/homebrew/bin` now by default, per https://brew.sh/ samples and:

```
❯ which cmake
/opt/homebrew/bin/cmake
```

**Review by DivineDominion — CHANGES_REQUESTED** [review]
Thanks for the changes! I believe the Homebrew path needs to be adjusted; Could you add an ARM architecture to the test matrix and check? Otherwise this looks good 👍

File: script/update_libgit2 (hunk 1 above)
Comment (hannesa2) [reply script/update_libgit2]: @DivineDominion  done

**Comment by DivineDominion** [comment]
@hannesa2 Thank you for the quick reaction!  You mentioned in the SPM issue that you're experienced with pipelines -- can we get GitHub CI to run on both Intel and arm64 to verify the PATH setting without too much trouble? To me it looks like we can't do that within this test matrix because we need different images,  i.e. `macos-13-xlarge`:
https://github.com/actions/runner-images?tab=readme-ov-file#available-images

But I'm not sure myself tbh

**Comment by hannesa2** [comment]
Yes I could do, but step by step.
I want to avoid to add more and more features to one pull request, which is never merged. (and focus an interest is lost)
So please merge _first_ this one, _than_ I can jump inst arm64 issue with an new pull request ....

**Comment by DivineDominion** [comment]
@hannesa2 Fair! While this PATH change may break, the old PATH wouldn't have worked, either, so let's do it this way.

**Comment by DivineDominion** [comment]
FWIW my Xcode 15 fixes from this PR still worked locally: https://github.com/SwiftGit2/SwiftGit2/pull/203/files#diff-745c5da16ef62f9f0fe570838efbdeebd58735407d9fef130b38749d5ee3d2f8

**Pull request merged**